cmd/osbuild-worker/jobimpl-bootc-manifest.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"regexp"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/osbuild/images/pkg/manifest"
+	"github.com/osbuild/osbuild-composer/internal/worker"
+	"github.com/osbuild/osbuild-composer/internal/worker/clienterrors"
+)
+
+type BootcManifestJobImpl struct{}
+
+type ErrorInvalidImageType string
+
+func (e ErrorInvalidImageType) Error() string { return string(e) }
+
+type ErrorCommandExecution string
+
+func (e ErrorCommandExecution) Error() string { return string(e) }
+
+type ErrorInvalidArguments string
+
+func (e ErrorInvalidArguments) Error() string { return string(e) }
+
+type ErrorUnknown string
+
+func (e ErrorUnknown) Error() string { return string(e) }
+
+func setResponseError(err error, result *worker.BootcManifestJobResult) {
+	switch e := err.(type) {
+	case ErrorInvalidImageType:
+		result.JobError = clienterrors.New(
+			clienterrors.ErrorInvalidImageType,
+			"Unsupported image type",
+			e.Error(),
+		)
+	case ErrorCommandExecution:
+		result.JobError = clienterrors.New(
+			clienterrors.ErrorBootcManifestExec,
+			"Error running bootc-image-builder command",
+			e.Error(),
+		)
+	default:
+		result.JobError = clienterrors.New(
+			clienterrors.ErrorInvalidManifestArgs,
+			"Invalid bootc parameters or parameter combination",
+			err.Error(),
+		)
+	}
+}
+
+func buildManifestCommand(args *worker.BootcManifestJob) *exec.Cmd {
+	baseArgs := []string{
+		"sudo", "podman", "run",
```

Comment:
Note that the worker runs as a root, so `sudo` is not needed for rootful. If we want to run podman rootless, we need to find a way to do it.

Reply:
I've been trying to enable bib for manifest generation in a rootless unprivileged  podman with no success so far, I asked @mvo5 for an advice here. A different solution would be to able to embed bib into the worker (just for the manifest part), or using a secure instance to run this rootful and privileged.